sbpy/utils/dastcom5.py

```diff
@@ -0,0 +1,575 @@
+        for line in inF:
+            if re.search(r"\b" + name.casefold() + r"\b", line.casefold()):
+                lines.append(line)
+    return lines
+
+
+def read_headers():
+    """Read `DASTCOM5` headers and return asteroid and comet headers.
+
+    Headers are two numpy arrays with custom dtype.
+
+    Returns
+    -------
+    ast_header, com_header : tuple (numpy.ndarray)
+        DASTCOM5 headers.
+
+    """
+
+    ast_path = os.path.join(DBS_LOCAL_PATH, "dast5_le.dat")
+    ast_dtype = np.dtype(
+        [
+            ("IBIAS1", np.int32),
+            ("BEGINP1", "|S8"),
+            ("BEGINP2", "|S8"),
+            ("BEGINP3", "|S8"),
+            ("ENDPT1", "|S8"),
+            ("ENDPT2", "|S8"),
+            ("ENDPT3", "|S8"),
+            ("CALDATE", "|S19"),
+            ("JDDATE", np.float64),
+            ("FTYP", "|S1"),
+            ("BYTE2A", np.int16),
+            ("IBIAS0", np.int32),
+        ]
+    )
+
+    with open(ast_path, "rb") as f:
+        ast_header = np.fromfile(f, dtype=ast_dtype, count=1)
+
+    com_path = os.path.join(DBS_LOCAL_PATH, "dcom5_le.dat")
+    com_dtype = np.dtype(
+        [
+            ("IBIAS2", np.int32),
+            ("BEGINP1", "|S8"),
+            ("BEGINP2", "|S8"),
+            ("BEGINP3", "|S8"),
+            ("ENDPT1", "|S8"),
+            ("ENDPT2", "|S8"),
+            ("ENDPT3", "|S8"),
+            ("CALDATE", "|S19"),
+            ("JDDATE", np.float64),
+            ("FTYP", "|S1"),
+            ("BYTE2C", np.int16),
+        ]
+    )
+
+    with open(com_path, "rb") as f:
+        com_header = np.fromfile(f, dtype=com_dtype, count=1)
+
+    return ast_header, com_header
+
+
+def read_record(record):
+    """Read `DASTCOM5` record and return body data.
+
+    Body data consists of numpy array with custom dtype.
+
+    Parameters
+    ----------
+    record : int
+        Body record.
+
+    Returns
+    -------
+    body_data : numpy.ndarray
+        Body information.
+
+    """
+    ast_header, com_header = read_headers()
+    ast_path = os.path.join(DBS_LOCAL_PATH, "dast5_le.dat")
+    com_path = os.path.join(DBS_LOCAL_PATH, "dcom5_le.dat")
+    # ENDPT1 indicates end of numbered asteroids records
+    if record <= int(ast_header["ENDPT2"][0].item()):
+        # ENDPT2 indicates end of unnumbered asteroids records
+        if record <= int(ast_header["ENDPT1"][0].item()):
+            phis_rec = 835 * (record - ast_header["IBIAS0"][0].item() - 1)
+        else:
+            phis_rec = 835 * (record - ast_header["IBIAS1"][0].item() - 1)
+
+        with open(ast_path, "rb") as f:
+            f.seek(phis_rec, os.SEEK_SET)
+            body_data = np.fromfile(f, dtype=AST_DTYPE, count=1)
+    else:
+        phis_rec = 976 * (record - com_header["IBIAS2"][0].item() - 1)
+        with open(com_path, "rb") as f:
+            f.seek(phis_rec, os.SEEK_SET)
+            body_data = np.fromfile(f, dtype=COM_DTYPE, count=1)
+    return body_data
+
+
+def download_dastcom5():
```

Comment:
update will mean removing the version user has and downloading completely newer version? I didn't understand.

Reply:
Yes, sorry if I was unclear. By 'update' I mean to overwrite the existing local version with the latest available version.

Reply:
> instead of downloading the files into ~/.sbpy, can you download them into data_archive in the sbpy directory

Won't the users installing it from pip find it difficult to access the database? I think it is fine if someone is cloning the repository somewhere and then runs `pip install -e .`. Otherwise in my opinion, it won't be good?

Reply:
Good point.  I don't think we should assume that the sbpy directory is writable.

Reply:
Ok, so will we use `~/.sbpy` as the location of the database? `~/sbpy/dastcom5/`?

Reply:
Yes @mommermi